File: problems/867/solution.js
/**
 * @param {number[][]} A
 * @return {number[][]}
 */
var transpose = function(A) {
    var transposed = [];
    
    for (var i = 0; i < A.length; i++) {
        for (var j = 0; j < A[i].length; j++) {
            if (!Array.isArray(transposed[j])) {
                transposed[j] = [];
            }
            
            transposed[j][i] = A[i][j];
        }
    }
    
    return transposed;
};

File: problems/867/solution.go
package main
func transpose(A [][]int) [][]int {
    r, c  := len(A), len(A[0])
    transposed := make([][]int, c)
    
    for i := 0; i < c; i++ {
        transposed[i] = make([]int, r)
        
        for j := 0; j < r; j++ {
            transposed[i][j] = A[j][i]
        }
    }
    
    return transposed
}